### PR TITLE
PMM-7 Remove redundant ci-default.yml

### PR DESCRIPTION
**Pull request by ademidoff**
https://jira.percona.com/browse/PMM-7

**Problem**
We are managing two similar data structures: one in `.gitmodules` and one in `ci-default.yml`. The second one is currently being used by `ci.py`, which combines it with the `config overrides` located in `ci.yml`. It turns out, we can still achieve the goal without `ci-default.yml`, since the Python script can read as well from the `.gitmodules` config file.

**Solution**
Get rid of `ci-default.yml`, which contains the same data as `.gitmodules`.

This PR does that and it also refactors the code of `ci.py`.

### CODE REVIEW

**Review by talhabinrizwan — APPROVED** [review]

**Comment by JNKPercona** [comment]
Server docker: perconalab/pmm-server-fb:PR-3774-8757353
Client docker: perconalab/pmm-client-fb:PR-3774-8757353
Client tarball: https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm-client/pmm-client-PR-3774-8757353.tar.gz
Staging instance: https://pmm.cd.percona.com/job/pmm3-aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-3774-8757353&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm-client/pmm-client-PR-3774-8757353.tar.gz

**Comment by JNKPercona** [comment]
API tests have failed: https://pmm.cd.percona.com/job/pmm3-api-tests/1501/

File: ci.py [hunk 1]
```diff
@@ -42,8 +44,17 @@ def read_config_override(self):
             return yaml.load(f, Loader=yaml.FullLoader)
 
     def read_config(self):
```
Comment (BupycHuk) [inline ci.py:46]: read_submodules?

**Review by BupycHuk — APPROVED** [review]

**Comment by JNKPercona** [comment]
Server docker: perconalab/pmm-server-fb:PR-3774-e6af5d5
Client docker: perconalab/pmm-client-fb:PR-3774-e6af5d5
Client tarball: https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm-client/pmm-client-PR-3774-e6af5d5.tar.gz
Staging instance: https://pmm.cd.percona.com/job/pmm3-aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-3774-e6af5d5&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm-client/pmm-client-PR-3774-e6af5d5.tar.gz

**Comment by JNKPercona** [comment]
API tests have failed: https://pmm.cd.percona.com/job/pmm3-api-tests/1568/

**Pull request merged**